scripts/app.js

```diff
@@ -332,10 +333,18 @@ function shackUp() {
 		var sanitizedListings = shack.sanitizeListings(jqXHR.data);
 		// Set total pages so that other requests don't ask for more data when there is none.
 		shack.totalPages = jqXHR.data.totalPages;
-		// Concatenate the current data to the currentItems cache of data
-		shack.currentItems = shack.currentItems.concat(sanitizedListings);
-		shack.showListings( { data : sanitizedListings } );
-		shack.initSwipe( $('.listing').last() );
+
+		if ( sanitizedListings.length ) {
+			// Concatenate the current data to the currentItems cache of data
+			shack.currentItems = shack.currentItems.concat(sanitizedListings);
+			shack.showListings( { data : sanitizedListings } );
+			shack.initSwipe( $('.listing').last() );
+		} else {
+			// show "no results" page here
+			$('.error__loading').hide();
+			alert('no listings match your search, please try again');
```

Comment:
Instead of an alert, can we create a div in the style of our other error messaging?